R/agyw-integration.R

```diff
@@ -0,0 +1,1659 @@
+      )
+
+    df <- dplyr::bind_rows(df, df_x)
+  }
+
+  # Adjusting country specific sexual debut estimates with age distribution of
+  # FSW from Thembisa
+  #Downloaded from: https://www.thembisa.org/content/downloadPage/Thembisa4_3
+  zaf_propensity <- naomi.resources::load_agyw_exdata("zaf_propensity", iso3 = "ZAF") %>%
+    dplyr::filter(kp == "FSW")
+
+  fsw_est <- df %>%
+    #  Add FSW propensity estimates from ZAF
+    dplyr::left_join(zaf_propensity, by = "age_group") %>%
+    # Calculate distribution of FSWs
+    dplyr::mutate(dist = eversexpop_prop * propensity) %>%
+    dplyr::group_by(area_id) %>%
+    dplyr::mutate(dist = dist / sum(dist)) %>%
+    dplyr::ungroup() %>%
+    # Add FSW PSEs
+    dplyr::full_join(
+      fsw %>% dplyr::select(total_fsw, iso3, area_id, area_level),
+      by = dplyr::join_by(area_id, iso3, area_level)
+    ) %>%
+    # Calculate FSW proportions
+    dplyr::mutate(
+      fsw = dist * total_fsw,
+      fsw_prop = fsw / population,
+      consensus_estimate = fsw_consensus
+    ) %>%
+    dplyr::select(-eversexpop, -eversexpop_prop, -propensity, -dist, -total_fsw)
+
+  fsw_est
+
+}
+
+
+#' Disaggregate admin1 PWID proportions from Oli's KP model to 5-age groups
+#'
+#' @param outputs Naomi output.
+#' @param options Naomi model options.
+#' @param naomi_pop Naomi population estimates for T2.
+#' @param kp_consensus Key pop consensus estimates.
+#'
+#' @return District level PWID estimates by 5-year age bands for ages 15-49.
+#' @keywords internal
+
+agyw_disaggregate_pwid <- function(outputs,
+                                   options,
+                                   naomi_pop,
+                                   kp_consensus)
+{
+
+  # Extract country specific national PWID PSEs
+  iso3 <- options$area_scope
+
+  pse <- naomi.resources::load_agyw_exdata("kp_estimates", iso3) %>%
+    dplyr::filter(kp == "PWID", indicator == "pse_prop")
+
+  pwid_pse <- pse %>%
+    dplyr::rename(prop_pwid = median) %>%
+    dplyr::select(-indicator,-lower,-upper)
+
+  age_groups <- c("Y015_019", "Y020_024", "Y025_029", "Y030_034",
+                  "Y035_039", "Y040_044", "Y045_049")
+
+  pwid <- pwid_pse %>%
+    dplyr::mutate(age_group = "Y015_049") %>%
+    dplyr::left_join(naomi_pop %>% dplyr::filter(sex == "male"),
+                     by = dplyr::join_by(iso3, area_id, age_group)) %>%
+    dplyr::mutate(total_pwid = population * prop_pwid) %>%
+    dplyr::select(iso3, area_id, total_pwid, age_group, area_level)
+
+  # Check for consensus estimate of MSM
+  pwid_consensus <- kp_consensus[kp_consensus$key_population == "PWID", ]$population_size
+
+  if(!is.na(pwid_consensus)){
+    # Check if consensus estimate is larger than age matched population denominator
+    pop <- naomi_pop[naomi_pop$area_level == 0 & naomi_pop$age_group == "Y015_049" & naomi_pop$sex == "male",]$population
+    prop_pop <- pwid_consensus / pop
+
+    if(prop_pop >= 0.05) {
+      t_("PWID_CONSENSUS_WARNING")
```

Comment:
This also doesn't go anywhere

Reply:
Have removed and fixed up test to check that this working.